### PR TITLE
ENG-2047 Bugfix: User unable to switch workflows after hitting back button in browser

### PR DESCRIPTION
**Pull request by agiron123**
## Describe your changes and why you are making these changes
This PR fixes a bug where user is unable to switch workflows after clicking back from workflow details page.

## Related issue number (if any)
ENG-2047

## Loom demo (if any)
https://www.loom.com/share/92f1f04d1e9b48fabfc435a86c5184ba

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/app/index.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ const App = () => {
   const pathPrefix = getPathPrefix();
   let routesContent: React.ReactElement;
   routesContent = (
-    <Routes>      
+    <Routes>
       <Route path={`${pathPrefix ?? "/"}`} element={<RequireAuth user={user}><HomePage user={user} /> </RequireAuth>} />
```
Comment (agiron123) [inline src/ui/app/index.tsx:34]: these changes can be checked out to main, will revert.

File: src/ui/common/src/components/pages/workflow/id/index.tsx [hunk 2]
```diff
@@ -80,17 +82,39 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
     (state: RootState) => state.workflowReducer.artifactResults[currentNode.id]
   );
 
+  const dagName = workflow.selectedDag?.metadata?.name;
+
+  useEffect(() => {
```
Comment (agiron123) [inline src/ui/common/src/components/pages/workflow/id/index.tsx:87]: remove this block, no longer needed.

File: src/ui/common/src/components/pages/workflow/id/index.tsx [hunk 3]
```diff
@@ -80,17 +82,39 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
...
   }, [workflow.selectedDag]);
 
+  const resetWorkflowState = () => {
+    console.log('resetWorkflowState()');
+    dispatch(resetState());
+  };
+
+  useEffect(() => {
+    window.onpopstate = () => {
+      console.log('inside onpopstate');
```
Comment (agiron123) [inline src/ui/common/src/components/pages/workflow/id/index.tsx:104]: remove log

File: src/ui/common/src/components/pages/workflow/id/index.tsx [hunk 4]
```diff
@@ -80,17 +82,39 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
...
+      resetWorkflowState();
+    };
+
+    dispatch(resetState());
```
Comment (agiron123) [inline src/ui/common/src/components/pages/workflow/id/index.tsx:108]: change to resetWorkflowState();

File: src/ui/common/src/components/pages/workflow/id/index.tsx [hunk 5]
```diff
@@ -427,9 +452,17 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
       breadcrumbs={[
         BreadcrumbLink.HOME,
         BreadcrumbLink.WORKFLOWS,
-        new BreadcrumbLink(path, workflow.selectedDag.metadata.name),
+        new BreadcrumbLink(path, dagName),
       ]}
       user={user}
+      onBreadCrumbClicked={() => {
+        console.log('onBreadcrumbclicked');
+        dispatch(resetState());
```
Comment (agiron123) [inline src/ui/common/src/components/pages/workflow/id/index.tsx:460]: resetWorkflowState()

File: src/ui/common/src/components/pages/workflow/id/index.tsx [hunk 6]
```diff
@@ -427,9 +452,17 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
...
+      }}
+      onSidebarItemClicked={() => {
+        console.log('onSidebarItemClicked');
```
Comment (agiron123) [inline src/ui/common/src/components/pages/workflow/id/index.tsx:463]: resetWorkflowState()

**Review by cw75 — APPROVED** [review]

**Pull request merged**